sunpy/net/jsoc/jsoc.py

```diff
@@ -650,27 +650,23 @@ def _make_recordset(self, series, start_time='', end_time='', wavelength='',
                                  "prime-keys is not yet supported. Alternative is to "
                                  "use only one of the primekey to query for series data.")
                 raise ValueError(error_message)
-            if match:
-                timestr = '{}'.format(primekey.pop(list(match)[0], ''))
-            else:
-                timestr = ''
-        if wavelength != '':
-            if not primekey.get('WAVELNTH', ''):
-                if isinstance(wavelength, list):
-                    wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
-                    wavelength = str(wavelength)
-                else:
-                    wavelength = f'{int(np.ceil(wavelength.to(u.AA).value))}'
+            timestr = f"{primekey.pop(list(match)[0], '')}" if match else ''
+        if wavelength == '':
+            # This is executed when wavelength has been passed through PrimeKey().
+            wavelength = f"{primekey.pop('WAVELNTH', '')}"
+        elif not primekey.get('WAVELNTH', ''):
+            if isinstance(wavelength, list):
+                wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
+                wavelength = str(wavelength)
```

Comment:
Looks like this block is missing a test, I guess adding one is your penalty for cleaning up this code :stuck_out_tongue:

Reply:
I have no idea who to even get into this block.

Reply:
So I deleted it.